web/src/settings_data.ts

```diff
@@ -224,7 +224,15 @@ export function user_can_create_user_groups(): boolean {
 }
 
 export function user_can_add_custom_emoji(): boolean {
-    return user_has_permission(realm.realm_add_custom_emoji_policy);
+    if (page_params.is_spectator) {
```

Comment:
This is not needed as we have this check in `user_has_permission_for_group_setting` now.